File: DataGridView ComboBox with ValueMember/[C#][VB]-DataGridView ComboBox with ValueMember/C# and VB/Example_vb/Classes/DataOperations.vb
Imports System.Data.SqlClient

Public Class DataOperations
    ''' <summary>
    ''' The connection is pointing to my SQL-Server instance, depending on your SQL-Server install you may need to change
    ''' Data Source=KARENS-PC to Data Source=SQLEXPRESS 
    ''' Here is a web page to assist with setting the connection up
    ''' https://www.connectionstrings.com/sql-server/
    ''' </summary>
    Private ConnectionString As String = "Data Source=KARENS-PC;Initial Catalog=ExampleDataGridViewComboBox_1;Integrated Security=True"
    Public Property PersonsTable As New DataTable
    Public Property ColorTable As New DataTable
    Public Property InformationTable As New DataTable
    Public Property ColorDictionary As New Dictionary(Of Integer, String)

    Public Property Exception As New ErrorInformation

    Public Sub GetData()
        Using cn As New SqlConnection With {.ConnectionString = ConnectionString}
            Using cmd As New SqlCommand With {.Connection = cn}
                Try
                    cn.Open()
                Catch ex As Exception
                    Exception.HasError = True
                    Exception.Message = ex.Message
                End Try


                cmd.CommandText = "SELECT ColorId,ColorText FROM Colors"
                ColorTable.Load(cmd.ExecuteReader)
                Try

                    For Each row As DataRow In ColorTable.Rows
                        ColorDictionary.Add(row.Field(Of Integer)("ColorId"), row.Field(Of String)("ColorText"))
                    Next
                Catch ex As Exception
                    Console.WriteLine(ex.Message)
                End Try

                cmd.CommandText = "SELECT Id,[Text] FROM Information"
                InformationTable.Load(cmd.ExecuteReader)
                cmd.CommandText = "SELECT Id,ColorId ,FirstName FROM Person"
                PersonsTable.Load(cmd.ExecuteReader)
                Console.WriteLine()
            End Using
        End Using
    End Sub
    Public Function UpdatePerson(ByVal PersonId As Integer, ColorId As Integer) As Boolean
        Dim Result As Integer = 0
        Using cn As New SqlConnection With {.ConnectionString = ConnectionString}
            Using cmd As New SqlCommand With {.Connection = cn}
                cmd.CommandText = "UPDATE Person SET ColorId = @ColorId WHERE Id = @PersonId"
                cmd.Parameters.AddWithValue("@ColorId", ColorId)
                cmd.Parameters.AddWithValue("@PersonId", PersonId)

                Try
                    cn.Open()
                Catch ex As Exception
                    Exception.HasError = True
                    Exception.Message = ex.Message
                End Try

                Try
                    Result = cmd.ExecuteNonQuery
                Catch ex As Exception
                    Console.WriteLine(ex.Message)
                End Try
            End Using
        End Using

        Return Result = 1

    End Function
End Class
Public Class ErrorInformation
    Public Property Message As String
    Public Property HasError As Boolean
End Class

File: DataGridView ComboBox with ValueMember/[C#][VB]-DataGridView ComboBox with ValueMember/C# and VB/Example_cs/Classes/DataOperations.cs
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Data;

namespace Example_cs.Classes
{
    public class DataOperations
    {
        /// <summary>
        /// The connection is pointing to my SQL-Server instance, depending on your SQL-Server install you may need to change
        /// Data Source=KARENS-PC to Data Source=SQLEXPRESS 
        /// Here is a web page to assist with setting the connection up
        /// 
        /// https://www.connectionstrings.com/sql-server/
        /// 
        /// </summary>
        private string ConnectionString = "Data Source=KARENS-PC;Initial Catalog=ExampleDataGridViewComboBox_1;Integrated Security=True";

        public DataTable PersonsTable = new DataTable();
        public DataTable ColorTable = new DataTable();
        public DataTable InformationTable = new DataTable();

        public Dictionary<int, string> ColorDictionary = new Dictionary<int, string>();

        public ErrorInformation Exception = new ErrorInformation();
        /// <summary>
        /// Gets data for DataGridView
        /// </summary>
        public void GetData()
        {
            using (SqlConnection cn = new SqlConnection { ConnectionString = ConnectionString })
            {
                using (SqlCommand cmd = new SqlCommand { Connection = cn })
                {
                    try
                    {
                        cn.Open();
                    }
                    catch (Exception ex)
                    {
                        Exception.HasError = true;
                        Exception.Message = ex.Message;
                    }


                    cmd.CommandText = "SELECT ColorId,ColorText FROM Colors";
                    ColorTable.Load(cmd.ExecuteReader());
                    try
                    {

                        foreach (DataRow row in ColorTable.Rows)
                        {
                            ColorDictionary.Add(row.Field<int>("ColorId"), row.Field<string>("ColorText"));
                        }
                    }
                    catch (Exception ex)
                    {
                        Console.WriteLine(ex.Message);
                    }

                    cmd.CommandText = "SELECT Id,[Text] FROM Information";
                    InformationTable.Load(cmd.ExecuteReader());
                    cmd.CommandText = "SELECT Id,ColorId ,FirstName FROM Person";
                    PersonsTable.Load(cmd.ExecuteReader());
                    Console.WriteLine();
                }
            }
        }
        /// <summary>
        /// Update a row in the Person table
        /// </summary>
        /// <param name="PersonId">Primary key for a specific record</param>
        /// <param name="ColorId">Key to the color to update</param>
        /// <returns></returns>
        public bool UpdatePerson(int PersonId, int ColorId)
        {
            int Result = 0;
            using (SqlConnection cn = new SqlConnection { ConnectionString = ConnectionString })
            {
                using (SqlCommand cmd = new SqlCommand { Connection = cn })
                {
                    cmd.CommandText = "UPDATE Person SET ColorId = @ColorId WHERE Id = @PersonId";
                    cmd.Parameters.AddWithValue("@ColorId", ColorId);
                    cmd.Parameters.AddWithValue("@PersonId", PersonId);

                    try
                    {
                        cn.Open();
                    }
                    catch (Exception ex)
                    {
                        Exception.HasError = true;
                        Exception.Message = ex.Message;
                    }

                    try
                    {
                        Result = cmd.ExecuteNonQuery();
                    }
                    catch (Exception ex)
                    {
                        Console.WriteLine(ex.Message);
                    }
                }
            }

            return Result == 1;

        }
    }
    /// <summary>
    /// Provides a container for DataOperations error information
    /// </summary>
    public class ErrorInformation
    {
        public string Message { get; set; }
        public bool HasError { get; set; }
    }
}
